src/App.tsx

```diff
@@ -50,15 +44,18 @@ export const App: React.FC = () => {
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector
+                  value={author}
+                  onChange={user => dispatch(setAuthor(user))}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
                 {!author && <p data-cy="NoSelectedUser">No user selected</p>}
 
-                {author && !loaded && <Loader />}
+                {author && !posts.loaded && <Loader />}
```

Comment:
Ensure that the `Loader` component is correctly imported and implemented. If the component is not functioning as expected, it might cause issues in displaying the loading state.